core/src/main/java/org/apache/accumulo/core/file/rfile/RFileOperations.java

```diff
@@ -53,13 +56,16 @@ public class RFileOperations extends FileOperations {
 
   private static final Collection<ByteSequence> EMPTY_CF_SET = Collections.emptySet();
 
-  private static RFile.Reader getReader(FileOptions options) throws IOException {
+  private static IndexedFileSKVIterator getReader(FileOptions options) throws IOException {
     CachableBuilder cb = new CachableBuilder()
         .fsPath(options.getFileSystem(), new Path(options.getFilename()), options.dropCacheBehind)
         .conf(options.getConfiguration()).fileLen(options.getFileLenCache())
         .cacheProvider(options.cacheProvider).readLimiter(options.getRateLimiter())
         .cryptoService(options.getCryptoService());
-    return new RFile.Reader(cb);
+    final RFile.Reader reader = new RFile.Reader(cb);
+    return Optional.ofNullable(options.getFence())
```

Comment:
```suggestion
    return options.getFence() == null ? reader : new FencedReader(reader, options.getFence());
```

Functionally equivalent, but doesn't create objects.